server/lib/contributors.ts

```diff
@@ -99,7 +99,7 @@ const contributorsQuery = `
     c."isIncognito" as "isIncognito",
     BOOL_OR(COALESCE((c."data" ->> 'isGuest') :: boolean, FALSE)) AS "isGuest",
     COALESCE(MAX(m.description), MAX(tiers.name)) AS "description",
-    COALESCE(sum(transactions.amount) / count(DISTINCT m.id), 0) AS "totalAmountDonated"
+    COALESCE(sum(transactions."amountInHostCurrency" - transactions."paymentProcessorFeeInHostCurrency" - transactions."hostFeeInHostCurrency") / count(DISTINCT m.id), 0) AS "totalAmountDonated"
```

Comment:
If we switch the logic to use the host currency, we should also update:
- The docs in https://github.com/opencollective/opencollective-api/blob/0a210b25d5ad8e9c4a567f9fb72e2793e81533af/server/graphql/v1/types.js#L489 and https://github.com/opencollective/opencollective-api/blob/0a210b25d5ad8e9c4a567f9fb72e2793e81533af/server/graphql/v2/object/Contributor.ts#L45 to say "in host currency".
- The frontend (e.g. [here](https://github.com/opencollective/opencollective-frontend/blob/508c2bd16947eab201c8dc84f850342638daa521/components/collective-page/sections/TopFinancialContributors.js#L28)), as all these amounts are always displayed using `collective.currency`. We're going to display the wrong information if we don't do this (e.g. `MXN 100` when it's `USD 100`).

Related to this last point, I would personally prefer to make sure we keep returning this amount in the collective currency. Since the goal of this metric is to be displayed on the collective page, I think they'll be confused if we show the amounts in the host currency. To achieve that, we can either rely on `netAmountInCollectiveCurrency` or convert all amounts after the GraphQL query. Another way could be to introduce an `amountV2` on contributors, like what we're doing for https://github.com/opencollective/opencollective/issues/5198.

Reply:
Good point. This weird currency bug only affects the fiscal-host, which is the only entity that will end up listing transactions in different currencies.
I think it is a good idea to keep displaying in the collective currency, but I'm more inclined to calculate the total amount based on the host amount, since this is the money that is actually stored.
I'll try to join the currency FX here and make it work.